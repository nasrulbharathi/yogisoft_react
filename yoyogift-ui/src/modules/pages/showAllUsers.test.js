import React from "react";
import { shallow, mount } from "enzyme";
import { ShowAllUsers } from "./ShowAllUsers";
import axiosWrapper from "../../apis/axiosCreate";
import { List } from "react-virtualized";

describe("should render ShowAllUsers", () => {
  it("should be rendered", () => {
    const wrapper = shallow(<ShowAllUsers />);
    expect(wrapper).toMatchSnapshot();
  });

  it("test fetchUsers ", async () => {
    const wrapper = shallow(<ShowAllUsers />);
    const instance = wrapper.instance();
    const data2 = [
      {
        id: 1,
      },
      {
        id: 2,
      },
    ];
    axiosWrapper.get = jest
      .fn()
      .mockImplementation(() => Promise.resolve({ data: data2 }));
    await instance.fetchUsers();
    expect(instance.state.users.length).toBeGreaterThanOrEqual(1);
  });
});
