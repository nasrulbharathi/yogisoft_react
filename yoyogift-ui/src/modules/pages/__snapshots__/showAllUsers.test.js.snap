// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ShowAllUsers 1`] = `ShallowWrapper {}`;

exports[`should render ShowAllUsers should be rendered 1`] = `ShallowWrapper {}`;
